### PR TITLE
feat(webapp): Improve README and use dotenv

### PR DESCRIPTION
**Pull request by andreasnuesslein**
Here we go, finally a tiny PR:

Throw out the env.js.template/env.js in favor of vue cli's built-in
dotenv support.

I do dislike step 6 but I'm not sure how neccessary the `public:` is for the docker-based setup?

Cheers

### CODE REVIEW

**Comment by nils-wisiol** [comment]
This decouples the webapp .env from the stack .env, am I right?

**Comment by andreasnuesslein** [comment]
> This decouples the webapp .env from the stack .env, am I right?

Hm.. I don' think it interferes/interacts with the stack `.env` as such. It mainly throws out the "env.js" logic that seemed like a confusion to me, especially since the stack reads default `.env` anyways.

**Comment by peterthomassen** [comment]
> This decouples the webapp .env from the stack .env, am I right?

Not really, as the webapp .env is populated from the stack .env (https://github.com/desec-io/desec-stack/pull/384/commits/d3fb6cf93ff97c7ad904911ac102720dde895576#diff-5707ed9f775bf3b8dfdd5083fb6a9643) and then evaluated using envsubst, pretty much as before. It's just that from inside the webapp, you don't need to import the env JavaScript file.

File: .env.default [hunk 1]
```diff
@@ -11,6 +11,7 @@ DESECSTACK_IPV6_ADDRESS=fda8:7213:9e5e:1::0642:ac10:0080
 DESECSTACK_WWW_CERTS=./certs
 
 # API-related
+DESECSTACK_API_BASE_URL='/api/v1/'
```
[the remark this answers is not in the document]
Comment (andreasnuesslein) [reply .env.default:14]: It's used here https://github.com/desec-io/desec-stack/pull/384/files#diff-5707ed9f775bf3b8dfdd5083fb6a9643R1

[the remark this answers is not in the document]
Comment (nils-wisiol) [reply .env.default:14]: right but where is that one used?

[the remark this answers is not in the document]
Comment (andreasnuesslein) [reply .env.default:14]: in the axios HTTP helper in a bunch of places https://github.com/desec-io/desec-stack/pull/384/files#diff-948141e20e8c82c3af1e15645d33c26bL164

**Comment by andreasnuesslein** [comment]
> The reason we didn't use process.env was that the code will be compiled and then deployed as static files. In that case, how would `process.env` be evaluated? Would it be substituted at compile-time?

Oh - I must've overlooked this before. This is a good question, I'm not sure. I'm assuming that it will not be substituted, intentionally, as the `.env` is by design something that is supposed to happen per-host ("per environment") and I would assume that once I deploy my compiled webapp to another machine I would still like to change the .env-parameters of that machine. 
I can check though, to be certrain

**Comment by nils-wisiol** [comment]
In my tests, it seems like compiling the webapp appropriately replaces the environment, i.e. it's the behavior that we want.

**Comment by andreasnuesslein** [comment]
> This way, the envsubst-bloat can be removed entirely. Would you want to update this PR accordingly?

I planned on doing that but I don't have the whole docker setup running here at all. It might be safer if you did it to see if it still works afterwards?

**Comment by nils-wisiol** [comment]
Sure, will do; however your changes to the README will still be very much appreciated! I'll ping you as soon as I fixed the env setup.

**Pull request closed without merging**